packages/network-controller/src/create-network-client.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  createAsyncMiddleware,
+  createScaffoldMiddleware,
+  JsonRpcEngine,
+  mergeMiddleware,
+  JsonRpcMiddleware,
+} from 'json-rpc-engine';
+import {
+  createBlockCacheMiddleware,
+  createBlockRefMiddleware,
+  createBlockRefRewriteMiddleware,
+  createBlockTrackerInspectorMiddleware,
+  createInflightCacheMiddleware,
+  createFetchMiddleware,
+  createRetryOnEmptyMiddleware,
+} from '@metamask/eth-json-rpc-middleware';
+import {
+  providerFromEngine,
+  providerFromMiddleware,
+  SafeEventEmitterProvider,
+} from '@metamask/eth-json-rpc-provider';
+import { createInfuraMiddleware } from '@metamask/eth-json-rpc-infura';
+import type { Hex } from '@metamask/utils/dist';
+import { PollingBlockTracker } from 'eth-block-tracker/dist';
```

Comment:
Nit: This is a mistake from the extension — we shouldn't be "reaching" inside of packages and importing files directly from packages, we should expect they are available via the `main` file. What do you think about:

``` suggestion
import type { Hex } from '@metamask/utils';
import { PollingBlockTracker } from 'eth-block-tracker';
```